src/pages/login/CertificateGenerateBtn.tsx

```diff
@@ -0,0 +1,92 @@
+import type { FC } from "react";
+import { useState } from "react";
+import PasswordModal from "./PasswordModal";
+import { Button, Icon } from "@canonical/react-components";
+
+interface Certs {
+  crt: string;
+  pfx: string;
+}
+
+interface Props {
+  onMacOS?: boolean;
+}
+
+const CertificateGenerateBtn: FC<Props> = ({ onMacOS }) => {
```

Comment:
```suggestion
const CertificateGenerateBtn: FC<Props> = ({ isPasswordRequired }) => {
```
In this component and on the password modal, probably a better suiting name.